connector/src/main/java/org/jboss/as/connector/subsystems/datasources/AbstractDataSourceAdd.java

```diff
@@ -206,6 +207,8 @@ void firstRuntimeStep(OperationContext context, ModelNode operation, ModelNode m
                 dataSourceService.getDriverInjector());
 
          // If the authentication context is defined, add the capability
+         // TODO ELYTRON_ENABLED is redundant as AUTHENTICATION_CONTEXT implies elytron
```

Comment:
can't it be like
- no security set => use current authentication context (although I'm not sure how it would work without elytron subsystem)
- security-domain set => use legacy security services
- username+password(credential) reference

? 
this would allow us to get rid of elytron-enabled attribute